### PR TITLE
Spring Boot Data JPA task!V1

### PR DESCRIPTION
**Pull request by TkachukTara**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -57,13 +57,20 @@
 			<artifactId>mapstruct</artifactId>
 			<version>1.5.5.Final</version>
 		</dependency>
+
```
Comment (okuzan) [inline pom.xml:60]:
```suggestion
```

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+databaseChangeLog:
+  - include:
+      file: db/changelog/changes/create-books-table.yaml
```
Comment (okuzan) [inline src/main/resources/db/changelog/db.changelog-master.yaml:3]:
```suggestion
      file: db/changelog/changes/create-books-table.yaml
      
```

File: src/main/resources/db/changelog/changes/create-books-table.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                  type: varchar(255)
+              - column:
+                  name: cover_image
+                  type: varchar(255)
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (okuzan) [inline src/main/resources/db/changelog/changes/create-books-table.yaml:48]:
```suggestion
                    nullable: false
                    
```

File: src/main/java/book/store/model/Book.java [hunk 4]
```diff
@@ -5,11 +5,17 @@
...
+import jakarta.persistence.Table;
 import java.math.BigDecimal;
 import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
 @Entity
 @Data
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
```
Comment (okuzan) [inline src/main/java/book/store/model/Book.java]:
```suggestion
@SQLDelete(sql = "UPDATE books SET is_deleted = TRUE WHERE id = ?")
@Where(clause = "is_deleted = FALSE")
```

**Review by okuzan — CHANGES_REQUESTED** [review]
Change spring.jpa.hibernate.ddl-auto property value to validate
PUT: /api/books/{id} (Update a specific book)

File: src/main/java/book/store/dto/UpdateBookRequestDto.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    private String title;
+    private String author;
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
+    private String isbn;
+
+    public UpdateBookRequestDto() {
+        this.isbn = "Isbn" + new Random().nextInt(1000);
+    }
```
Comment (Rommelua) [inline src/main/java/book/store/dto/UpdateBookRequestDto.java]: delete this constructor

File: src/main/java/book/store/service/impl/BookServiceImpl.java [hunk 6]
```diff
@@ -36,4 +37,16 @@ public BookDto findById(Long id) {
...
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto updateBook(Long id, UpdateBookRequestDto updatedBook) {
+        Book bookFromDb = bookMapper.toModel(updatedBook);
+        bookFromDb.setId(id);
```
Comment (Rommelua) [inline src/main/java/book/store/service/impl/BookServiceImpl.java]: check if id exist in DB

File: src/main/java/book/store/service/impl/BookServiceImpl.java [hunk 7]
```diff
@@ -36,4 +36,16 @@ public BookDto findById(Long id) {
...
     }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto updateBook(Long id, CreateBookRequestDto updatedBook) {
+        Book bookFromDb = bookMapper.toModel(updatedBook);
```
Comment (Rommelua) [inline src/main/java/book/store/service/impl/BookServiceImpl.java]: Why bookFrom**Db** ?

File: src/main/java/book/store/service/impl/BookServiceImpl.java (hunk 6 above)
Comment (Rommelua) [reply src/main/java/book/store/service/impl/BookServiceImpl.java]: not fixed

**Review by Rommelua — CHANGES_REQUESTED** [review]

**Pull request merged**